apis/Google.Cloud.Diagnostics.AspNetCore/Google.Cloud.Diagnostics.AspNetCore/Trace/CloudTraceExtension.cs

```diff
@@ -121,7 +124,15 @@ public static void UseGoogleTrace(this IApplicationBuilder app)
         /// </summary>
```

Comment:
Ah yes, you're right. At least I think you are. I find all of this very confusing.
An alternative would be make IManagedTracer singleton-like and make it find the right trace behind the scenes. We should definitely get other ASP.NET Core folks to look at this...